model/src/main/scala/lucuma/schemas/enums/DynamicEnums.scala

```diff
@@ -5,9 +5,19 @@ package lucuma.schemas.enums
 
 import io.circe.*
 import io.circe.parser.*
+import lucuma.core.util.Enumerated
 
 object DynamicEnums:
   val parsedEnums: ACursor =
     parse(Globals.enumMetadataString) match
       case Left(err)   => err.printStackTrace; throw err
       case Right(json) => json.hcursor
+
+  def enumeratedInstance[E](
+    field: String,
+    tagFn: E => String
+  )(using Decoder[E]): Enumerated[E] =
+    val values: List[E] = parsedEnums.downField(field).as[List[E]] match
+      case Left(err)   => err.printStackTrace; throw err
+      case Right(list) => list
```

Comment:
Nitpicking: you can omit constructing the `values` and return the `Enumerated` instance directly from the `Right` case.

Reply:
Good point